core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -405,4 +405,90 @@ class TSqlExpressionBuilder(functionBuilder: FunctionBuilder)
     val dataType = dataTypeBuilder.build(ctx.dataType())
     ir.Cast(expression, dataType, returnNullOnError = ctx.TRY_CAST() != null)
   }
+
+  override def visitJsonArray(ctx: JsonArrayContext): ir.Expression = {
+    val elements = buildExpressionList(Option(ctx.expressionList()))
+    val absentOnNull = checkAbsentNull(ctx.jsonNullClause())
+    buildJsonArray(elements, absentOnNull)
+  }
+
+  override def visitJsonObject(ctx: JsonObjectContext): ir.Expression = {
+    val namedStruct = buildNamedStruct(Option(ctx.jsonKeyValue().asScala.toList))
+    val absentOnNull = checkAbsentNull(ctx.jsonNullClause())
+    buildJsonObject(namedStruct, absentOnNull)
+  }
+
+  private def checkAbsentNull(ctx: JsonNullClauseContext): Boolean = {
+    ctx match {
+      case null => true
+      case clause if clause.ABSENT() == null => false
+      case _ => true
+    }
+  }
+
+  private def buildNamedStruct(ctx: Option[List[JsonKeyValueContext]]): ir.NamedStruct = {
+    val (keys, values) = ctx
+      .getOrElse(List.empty)
+      .map { keyValueContext =>
+        val expressions = keyValueContext.expression().asScala.toList
+        (expressions.head.accept(this), expressions(1).accept(this))
+      }
+      .unzip
+
+    ir.NamedStruct(keys, values)
+  }
+
+  private def buildExpressionList(ctx: Option[ExpressionListContext]): Seq[ir.Expression] = {
+    ctx.map(_.expression().asScala.map(_.accept(this))).getOrElse(Seq.empty)
+  }
+
+  /**
+   * Databricks SQL does not have a native JSON_ARRAY function, so we use a Lambda filter and TO_JSON instead, but have
+   * to cater for the case where an expression is NULL and the TSql option ABSENT ON NULL is set. When ABSENT ON NULL is
+   * set, then any NULL expressions are left out of the JSON array.
+   *
+   * @param args
+   *   the list of expressions yield JSON values
+   * @param absentOnNull
+   *   whether we should remove NULL values from the JSON array
+   * @return
+   *   IR for the JSON_ARRAY function
+   */
+  private[tsql] def buildJsonArray(args: Seq[ir.Expression], absentOnNull: Boolean): ir.Expression = {
+    if (absentOnNull) {
+      val lambdaVariable = ir.UnresolvedNamedLambdaVariable(Seq("x"))
```

Comment:
Currently, acceptance tests verify that the output tree doesn't contain any node with a class name starting with "Unresolved". Going forward in improving acceptance tests success rate, we'll have to either rename `UnresolvedNamedLambdaVariable` to, say, `NotYetResolvedNamedLambdaVariable` or, better, make acceptance tests more specific about the `Unresolved*` classes they don't allow in output trees.

Reply:
Yes - I also spotted that. But this is pre-declared IR, so I could not rename it.